d3a_api_client/websocket_device.py

```diff
@@ -16,22 +16,26 @@ def __init__(self, rest_client):
         self.command_response_buffer = []
 
     def received_message(self, message):
-        if "event" in message:
-            if message["event"] == "market":
-                self.client._on_market_cycle(message)
-            elif message["event"] == "tick":
-                self.client._on_tick(message)
-            elif message["event"] == "trade":
-                self.client._on_trade(message)
-            elif message["event"] == "finish":
-                self.client._on_finish(message)
-            else:
-                logging.error(f"Received message with unknown event type: {message}")
-        elif "command" in message:
-            self.command_response_buffer.append(message)
+        try:
+            if "event" in message:
+                if message["event"] == "market":
+                    self.client._on_market_cycle(message)
+                elif message["event"] == "tick":
+                    self.client._on_tick(message)
+                elif message["event"] == "trade":
+                    self.client._on_trade(message)
+                elif message["event"] == "finish":
+                    self.client._on_finish(message)
+                else:
+                    logging.error(f"Received message with unknown event type: {message}")
+            elif "command" in message:
+                self.command_response_buffer.append(message)
 
-        if "event" in message or "command" in message:
-            self.client._on_event_or_response(message)
+            if "event" in message or "command" in message:
+                self.client._on_event_or_response(message)
```

Comment:
Can you please add these 2 lines of code to the Aggregator websocket
```d3a_api_client/aggregator.py:16```

Reply:
Darn I forgot that this method is overridden. Will do that, thanks for pointing it out!